lib/miq_automation_engine/engine/miq_ae_service_model_base.rb

```diff
@@ -273,9 +283,7 @@ def ar_method(&block)
   end
 end
 
-# Register all of the service models for autoload on first use
-Dir.glob("#{File.dirname(__FILE__)}/../service_models/miq_ae_service_*.rb").each do |f|
+Dir.glob(MiqAeMethodService::MiqAeServiceModelBase::SERVICE_MODEL_GLOB).each do |f|
   f = File.basename(f, '.*')
-  const = f.split('-').map(&:camelize).join('_')
-  MiqAeMethodService.autoload(const, "service_models/#{f}")
+  MiqAeMethodService.autoload(MiqAeMethodService::MiqAeServiceModelBase.model_name_from_file(f), "service_models/#{f}")
```

Comment:
Doubling up on the basename sounds safe, and this is otherwise just an Extract Method, so it seems fine to me.